refactor: be more permissive in function inputs

EricCrosson opened the pull request:
Do not require the caller to pass owned strings, since a reference
will suffice.

EricCrosson committed:
refactor: be more permissive in function inputs

Do not require the caller to pass owned strings, since a reference
will suffice.

EricCrosson commented:
I'm still new to Rust, so I hope this is somewhat idiomatic and definitely useful, but I'm very open to feedback!

If this does turn out to be a desirable change I am happy to update the linear/spot files to match.

jukanntenn commented:
Actually I also new to rust. I checked rust docs about `AsRef<T>`, it seems an appropriate usage here. Feel free to make the changes.

EricCrosson committed:
refactor: replace to_string with to_owned

to make emphasize that the intent behind this function call is to
create a copy of the data for ownership, and that it is not the casee
we want to coerce some non-string into a `String`

EricCrosson commented:
Sounds good @jukanntenn, I've updated all uses of `&Vec<String>`
